### PR TITLE
Enhance embargo area

### PR DESCRIPTION
**Pull request by jh-RLI**
## Summary of the discussion

Describe the findings of the discussion in the issue or meeting.

### Changes

- Enhance embargo area feature (UI & API) [(#1804)](https://github.com/OpenEnergyPlatform/oeplatform/pull/1804)
  - The REST API raises an error when creating a table with embargo period but unsupported date time for start/end data
  - Update the table creation http "put" endpoint to handle embargo periods
 
### Automation

Closes #

### PR-Assignee

- [x] 🐙 Follow the workflow in [CONTRIBUTING.md](https://github.com/OpenEnergyPlatform/oeplatform/blob/develop/CONTRIBUTING.md)
- [x] 📝 Update the [CHANGELOG.md](https://github.com/OpenEnergyPlatform/oeplatform/blob/develop/versions/changelogs/current.md)
- [ ] 📙 Update the documentation on [mkdocs](https://openenergyplatform.github.io/oeplatform/)

### Reviewer

- [x] 🐙 Follow the [Reviewer Guidelines](https://github.com/rl-institut/super-repo/blob/develop/CONTRIBUTING.md#40-let-someone-else-review-your-pr)
- [x] 🐙 Provided feedback and show sufficient appreciation for the work done

### CODE REVIEW

File: api/views.py [hunk 1]
```diff
@@ -483,48 +483,109 @@ def __create_table(
...
             )
+
+            table_object.save()
+
+            if metadata:
+                actions.set_table_metadata(
+                    table=table, schema=schema, metadata=metadata, cursor=cursor
+                )
+
         if embargo_data:
```
Comment (wingechr) [inline api/views.py:513]: should be 
```python
else:  # has embargo_data
```

Comment (jh-RLI) [reply api/views.py:513]: I agree

**Comment by jh-RLI** [comment]
I changed the code ... it was a bit of a mess and i figured that keep doing this qick and dirty is no solution:

1. Now an embargo can be added (optionally) on table createn (http put):


If the pust request contains a payload like:

# Will raise an api error
"embargo": {"start": "2024-08-27", "end": "2024-02-27"},  # not implemented yet 
# works
"embargo": {"duration": "6_months"},
# works, will set no embargo
"embargo": None,

If no embargo payload is available this will also work. 

Full example:

```
{
    "columns": [
        # NOTE: first column should be numerical column named `id` .
        # Use `bigserial` if you want the database to create the re
        {"name": "id", "data_type": "bigserial", "primary_key": True},
        {"name": "name", "data_type": "varchar(18)", "is_nullable": False},
        {"name": "is_active", "data_type": "boolean"},
        {"name": "capacity_mw", "data_type": "float"},
        {"name": "installation_datetime_utc", "data_type": "datetime"},
    ],
    "constraints": [{"constraint_type": "PRIMARY KEY", "constraint_parameter": "id"}],
    "embargo": {"duration": "6_months"},
}
```

2. Move api is still availabe & movePublish api is new and handles embargo period:

api: /schema/{schema}/tables/{table}/move_publish/{destination}/

payload:

table_schema = {
    "embargo": {"duration": "1_year"},
}

**Comment by jh-RLI** [comment]
One thing is still missing: The movePublish api does not include a licence check ... Tables with or without a properotary licence can be published.

**Comment by jh-RLI** [comment]
@wingechr if you find the time could you check this again? This changes the table creation API a bit - I tested it but didn't implement the test because they cost me more time than I expected :/

File: api/tests/test_table.py [hunk 2]
```diff
@@ -286,3 +286,87 @@ def test_simple(self):
...
+        # move back
+        self.api_req(
+            "post",
+            schema=self.target_schema,
+            path=f"move/{self.test_schema}/",
+            exp_code=200,
+        )
+
+
+class TestMovePublish(APITestCaseWithTable):
```
Comment (wingechr) [inline api/tests/test_table.py:316]: @jh-RLI this is my test case now. hope it makes sense to you

Comment (jh-RLI) [reply api/tests/test_table.py:316]: <3

**Pull request merged**